inc/datahandlers/user.php

```diff
@@ -324,14 +324,78 @@ function verify_website()
 	 */
 	function verify_icq()
 	{
+		global $mybb;
+
 		$icq = &$this->data['icq'];
 
+		if($mybb->settings['allowicqfield'] == '' || !is_member($mybb->settings['allowicqfield'], getuser($this->data['uid'])))
+		{
+			$icq = '';
+			return true;
+		}
+
 		if($icq != '' && !is_numeric($icq))
 		{
 			$this->set_error("invalid_icq_number");
 			return false;
 		}
 		$icq = (int)$icq;
+
+		if(my_strlen($icq) > 10)
+		{
+			$this->set_error("contact_field_icqerror");
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Verifies if a Google Hangouts number is valid or not.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_google()
+	{
+		global $mybb;
+
+		$user = &$this->data;
+
+		if($mybb->settings['allowgooglefield'] == '' || !is_member($mybb->settings['allowgooglefield'], getuser($this->data['uid'])))
+		{
+			$this->data['google'] = '';
+			return true;
+		}
+
+		if(my_strlen($user['google']) > 75)
+		{
+			$this->set_error("contact_field_error");
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Verifies if a Skype number is valid or not.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_skype()
+	{
+		global $mybb;
+
+		$user = &$this->data;
+
+		if($mybb->settings['allowskypefield'] == '' || !is_member($mybb->settings['allowskypefield'], getuser($this->data['uid'])))
+		{
+			$this->data['skype'] = '';
```

Comment:
What about `$user['skype'] = '';`? You are being inconsistent in the same PR. Just pointing it out, it isn't something that would stop me from accepting a PR.